src/Core/Components/List/FluentAutocomplete.razor.cs

```diff
@@ -216,13 +230,15 @@ protected async Task InputHandlerAsync(ChangeEventArgs e)
         var args = new OptionsSearchEventArgs<TOption>()
         {
             Items = Items ?? Array.Empty<TOption>(),
-            Text = _valueText,
+            Text = ValueText,
         };
 
         await OnOptionsSearch.InvokeAsync(args);
 
-        Items = args.Items.Take(MaximumOptionsSearch);
-        SelectableItem = Items.FirstOrDefault();
+        Items = args.Items?.Take(MaximumOptionsSearch);
```

Comment:
With this code, `args.Items` cannot be null:

```csharp
?? Array.Empty<TOption>()
```

Reply:
If the user supplied callback for OnOptionsSearch sets e.Items to null, then it will blow up (I ran into that exception while playing with this PR)

Reply:
Oh... OK, if the user updates this property from `OnOptionsSearch.InvokeAsync`.
Make sense.